src/ops_complex.rs

```diff
@@ -605,15 +600,25 @@ pub fn generate_permutation_matrix(
         32,
         stream,
     )?;
+    let (unique_variable_indexes, temp_storage) = temp_storage.split_at_mut(num_bytes);
+    let unique_variable_indexes = unsafe { unique_variable_indexes.transmute_mut() };
+    let (run_lengths, temp_storage) = temp_storage.split_at_mut(num_bytes);
+    let run_lengths = unsafe { run_lengths.transmute_mut() };
+    let (num_runs_out, encode_temp_storage) = temp_storage.split_at_mut(cell_size);
+    let num_runs_out = unsafe { &mut num_runs_out.transmute_mut()[0] };
     set_to_zero(run_lengths, stream)?;
     encode(
```

Comment:
I hate to say this, but I want to suggest it as due diligence:
encode() needs several simultaneous large args. We might reduce the allocation high-water mark here if we "manually" broke RLE into a sequence of primitives (scan, filter, etc). Your call if you think that's worth the trouble.

Reply:
Hmm, we already reduced the requirement a lot and this would require some more thinking and refactoring.
I don't want to delay this PR anymore so i would leave it for next time.